src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CompositeModelBinder.cs

```diff
@@ -97,7 +97,17 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
                 var result = await binder.BindModelAsync(bindingContext);
                 if (result != null)
                 {
-                    return result;
+                    // Use returned ModelBindingResult if it either indicates model was set or is related to errors.
+                    // The second condition is necessary because some binders add errors that would be duplicated when
```

Comment:
:-1: if there's any `ModelState`, either an error occurred or one of the binders successfully added a value but oddly returned a !IsModelSet value.